packages/openactive-integration-tests/test/helpers/flow-stages/fetch-opportunities.js

```diff
@@ -0,0 +1,170 @@
+const _ = require('lodash');
+const config = require('config');
+const { generateUuid } = require('../generate-uuid');
+const { FlowStage } = require('./flow-stage');
+const { OpportunityFeedUpdateFlowStage } = require('./opportunity-feed-update');
+
+/**
+ * @typedef {import('chakram').ChakramResponse} ChakramResponse
+ * @typedef {import('../../types/OpportunityCriteria').OpportunityCriteria} OpportunityCriteria
+ * @typedef {import('../request-helper').RequestHelperType} RequestHelperType
+ * @typedef {import('../logger').BaseLoggerType} BaseLoggerType
+ * @typedef {import('./opportunity-feed-update').OrderItem} OrderItem
+ */
+
+/**
+ * @typedef {{
+ *   testInterfaceOpportunities: ChakramResponse[],
+ *   opportunityFeedExtractResponses: ChakramResponse[],
+ *   orderItems: OrderItem[],
+ * }} FetchOpportunitiesResponse
+ */
+
+const USE_RANDOM_OPPORTUNITIES = config.get('useRandomOpportunities');
+const SELLER_CONFIG = config.get('sellers');
+
+/**
+ * For each of the Opportunity Criteria, get or create (using https://openactive.io/test-interface/#post-test-interfacedatasetstestdatasetidentifieropportunities)
+ * an opportunity that matches the criteria from the [test interface](https://openactive.io/test-interface/).
+ *
+ * The responses only contain minimal info like ID, so `runOpportunityFeedUpdate()`
+ * needs to be called afterwards to actually get the full data of each opportunity.
+ *
+ * @param {object} args
+ * @param {OpportunityCriteria[]} args.orderItemCriteriaList
+ * @param {RequestHelperType} args.requestHelper
+ * @returns {Promise<ChakramResponse[]>}
+ */
+async function getOrCreateTestInterfaceOpportunities({ orderItemCriteriaList, requestHelper }) {
```

Comment:
The code for this function has mostly been copied from RequestState.fetchOpportunities